packages/shared/src/components/typography/Typography.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { HTMLAttributes, ReactElement, RefAttributes } from 'react';
+import classed from '../../lib/classed';
+
+export enum TypographyElement {
+  TIME = 'time',
+  P = 'p',
+  H1 = 'h1',
+  H2 = 'h2',
+  H3 = 'h3',
+  H4 = 'h4',
+  H5 = 'h5',
+  H6 = 'h6',
+}
+export enum TypographyType {
+  Caption2 = 'typo-caption2',
+  Caption1 = 'typo-caption1',
+  Footnote = 'typo-footnote',
+  Subhead = 'typo-subhead',
+  Callout = 'typo-callout',
+  Body = 'typo-body',
+  Title3 = 'typo-title3',
+  Title2 = 'typo-title2',
+  Title1 = 'typo-title1',
+  LargeTitle = 'typo-large-title',
+  Mega3 = 'typo-mega3',
+  Mega2 = 'typo-mega2',
+  Mega1 = 'typo-mega1',
+  Giga3 = 'typo-giga3',
+  Giga2 = 'typo-giga2',
+  Giga1 = 'typo-giga1',
+  Tera = 'typo-tera',
+}
+
+export enum TypographyColor {
+  Primary = 'text-text-primary',
+  Secondary = 'text-text-secondary',
+  Tertiary = 'text-text-tertiary',
+  Quaternary = 'text-text-quaternary',
+  Disabled = 'text-text-disabled',
+  Link = 'text-text-link',
+}
+
+type AllowedTags = keyof Pick<JSX.IntrinsicElements, TypographyElement>;
+type AllowedElements = HTMLTimeElement | HTMLParagraphElement;
+export type TypographyProps<Tag extends AllowedTags> = {
+  element?: TypographyElement;
+  type?: TypographyType;
+  color?: TypographyColor;
+  bold?: boolean;
+  ref?: RefAttributes<AllowedElements>['ref'];
+} & HTMLAttributes<AllowedElements> &
+  JSX.IntrinsicElements[Tag];
+
+export function Typography<TagName extends AllowedTags>({
+  element = TypographyElement.P,
```

Comment:
Agreed 👍

Reply:
I think we generally don't use it often, but when we do use `Tag`
Not sure what you prefer?

I'm happy with suggested `as` :)

Reply:
Actually renaming it `tag` seemed more inline, do let me know if you agree both :)

Reply:
Seems we use `tag` more, `as` is probably used in MUI and Next.js so thats why it was in my head. From my side its good 👍

Reply:
@rebelchris  No blocking opinion:
Still prefer `as` because it is used all around by every library, but I think consistency is the key for a project. If we already used `tag`/`element` in the past is ok to keep it! 👍

Reply:
Yeah let's stick to `tag` just because it's used already.
We can always down the line move all that to `as`?

Just more of a follow up for my idea:
how do they handle the element type then?

is it something like this?
```
export enum TypographyAs {
  Time = 'time',
  P = 'p',
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
}
```

Reply:
I think they use union type `p | h1 | h2`... (quicker to type in the code 😆 ), but enums are cool for me as well.